torch policy now includes model.metrics

holli opened the pull request:
## Why are these changes needed?

I'm playing around with pytorch and imitation loss in model but I couldn't get custom metrics to be shown anywhere. If I understood correctly the policy should handle saving custom model.metrics. [torch_policy.py](https://github.com/ray-project/ray/blob/master/rllib/policy/torch_policy.py) does not have any references to model.metrics where as [tf_policy.py#L259](https://github.com/ray-project/ray/blob/master/rllib/policy/tf_policy.py#L259) has. I copied similar to pytorch_policy so that it works in my project.

pytorch_policy custom_loss support was added only recently in https://github.com/ray-project/ray/issues/8507 / https://github.com/ray-project/ray/pull/9142 so I guess this change was just missing from that.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

ericl commented:
Looks like this is breaking the following test:

//rllib:examples/custom_loss_torch                                       FAILED in 3 out of 3 in 26.0s

https://travis-ci.com/github/ray-project/ray/jobs/372340812

holli commented:
Oh cool so that example was used as a test. Lets try again.

I fixed that test. It was broken but not noticed as metrics() was never called. I also changed the metrics to use plain scalars instead of torch tensors. Previously metrics included tensors with gradient information and it didn't feel clean.
